File: my-app/src/types/match.ts
export interface MatchResult {
  id: number;
  matchId: number;
  homeScore: number;
  awayScore: number;
  homePenaltyScore?: number;
  awayPenaltyScore?: number;
  winningTeamId?: number;
  winningTeamName?: string;
  isDraw: boolean;
  note?: string;
  detailJson?: string;
  created?: string;
}

export interface Match {
  id: number;
  matchCode?: string;
  round?: string;
  scheduledStartTime?: string;
  actualStartTime?: string;
  endTime?: string;
  location?: string;
  status: string;
  tournamentSportId: number;
  sportName?: string;
  tournamentName?: string;
  groupId?: number;
  groupName?: string;
  homeTeamId?: number;
  homeTeamName?: string;
  awayTeamId?: number;
  awayTeamName?: string;
  result?: MatchResult;
  created?: string;
}

export interface CreateUpdateMatch {
  matchCode?: string;
  round?: string;
  scheduledStartTime?: string;
  actualStartTime?: string;
  endTime?: string;
  location?: string;
  status?: string;
  tournamentSportId: number;
  groupId?: number;
  homeTeamId?: number;
  awayTeamId?: number;
}

export interface CreateUpdateMatchResult {
  matchId: number;
  homeScore: number;
  awayScore: number;
  homePenaltyScore?: number;
  awayPenaltyScore?: number;
  winningTeamId?: number;
  isDraw?: boolean;
  note?: string;
  detailJson?: string;
}

File: API/Dms.Domain/Entities/Match.cs
using Dms.Domain.Common;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Dms.Domain.Entities
{
    public class Match : BaseEntity
    {
        public string? MatchCode { get; set; }
        public string? Round { get; set; } // Vòng bảng, Tứ kết, Bán kết, Chung kết...
        public DateTime? ScheduledStartTime { get; set; }
        public DateTime? ActualStartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public string? Location { get; set; } // Sân thi đấu / Nhà thi đấu
        public string Status { get; set; } = "Scheduled"; // Scheduled, InProgress, Completed, Postponed, Cancelled

        [ForeignKey(nameof(TournamentSportId))]
        public int TournamentSportId { get; set; }
        public virtual TournamentSport TournamentSport { get; set; } = null!;

        [ForeignKey(nameof(GroupId))]
        public int? GroupId { get; set; }
        public virtual Group? Group { get; set; }

        // Đội nhà / Đội 1
        [ForeignKey(nameof(HomeTeamId))]
        public int? HomeTeamId { get; set; }
        public virtual Team? HomeTeam { get; set; }

        // Đội khách / Đội 2
        [ForeignKey(nameof(AwayTeamId))]
        public int? AwayTeamId { get; set; }
        public virtual Team? AwayTeam { get; set; }

        // Kết quả trận đấu (1 - 1 với MatchResult)
        public virtual MatchResult? Result { get; set; }
    }
}
